packages/tunnel/src/commands/deploy/index.ts

```diff
@@ -0,0 +1,106 @@
+import { existsSync } from 'node:fs';
+import path from 'node:path';
+
+import { isValidUrl } from '@logto/core-kit';
+import chalk from 'chalk';
+import ora from 'ora';
+import type { CommandModule } from 'yargs';
+
+import { consoleLog } from '../../utils.js';
+
+import { type DeployCommandArgs } from './types.js';
+import { deployToLogtoCloud } from './utils.js';
+
+const tunnel: CommandModule<unknown, DeployCommandArgs> = {
+  command: ['deploy'],
+  describe: 'Deploy your custom UI assets to Logto Cloud',
+  builder: (yargs) =>
+    yargs
+      .options({
+        auth: {
+          describe: 'Auth credentials of your Logto M2M application. E.g.: <app-id>:<app-secret>',
+          type: 'string',
+        },
+        endpoint: {
+          describe:
+            'Logto endpoint URI that points to your Logto Cloud instance. E.g.: https://<tenant-id>.logto.app/',
+          type: 'string',
+        },
+        path: {
+          alias: ['experience-path'],
+          describe: 'The local folder path of your custom sign-in experience assets.',
+          type: 'string',
+        },
+        resource: {
+          alias: ['management-api-resource'],
+          describe: 'Logto Management API resource indicator. Required if using custom domain.',
+          type: 'string',
+        },
+        verbose: {
+          describe: 'Show verbose output.',
+          type: 'boolean',
+          default: false,
+        },
+      })
+      .epilog(
+        `Refer to our documentation for more details:\n${chalk.blue(
+          'https://docs.logto.io/docs/references/tunnel-cli/deploy'
+        )}`
+      ),
+  handler: async (options) => {
+    const {
+      auth,
+      endpoint,
+      path: experiencePath,
+      resource: managementApiResource,
+      verbose,
+    } = options;
+    if (!auth) {
+      consoleLog.fatal(
+        'Must provide valid Machine-to-Machine (M2M) authentication credentials. E.g. `<app-id>:<app-secret>`'
```

Comment:
Maybe add some instructions to let user know how to provide these values.
```suggestion
        'Must provide valid Machine-to-Machine (M2M) authentication credentials. E.g. add `auth` options in the command line `--auth <app-id>:<app-secret>`' or add it in the env file
```